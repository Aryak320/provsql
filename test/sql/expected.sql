\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE expected_result AS
SELECT city, expected(COUNT(*)) AS c1, expected(COUNT(id)) AS c2, expected(SUM(id)) AS s
FROM personnel
GROUP BY CITY;

SELECT remove_provenance('expected_result');

SELECT city, ROUND(c1::numeric,2) AS c1, ROUND(c2::numeric,2) AS c2, ROUND(s::numeric,2) AS s
FROM expected_result
ORDER BY city;

DROP TABLE expected_result;

-- Non-supported
SELECT expected('toto'::text) FROM personnel;
SELECT expected(MAX(id)) FROM personnel;
SELECT expected(AVG(id)) FROM personnel;
