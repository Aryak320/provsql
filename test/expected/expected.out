\set ECHO none
remove_provenance

(1 row)
city|c1|c2|s|min|max
Berlin|1.10|1.10|6.50|Infinity|-Infinity
New York|0.30|0.30|0.50|Infinity|-Infinity
Paris|1.40|1.40|7.00|Infinity|-Infinity
(3 rows)
remove_provenance

(1 row)
city|c1|c2|s|min|max
Berlin|1.34|1.34|7.93|5.54|6.56
New York|1.07|1.07|1.79|1.64|1.71
Paris|1.63|1.63|8.14|4.55|5.56
(3 rows)
ERROR:  invalid input syntax for agg_token: "toto"
ERROR:  Cannot compute expected value for aggregation function avg
